static/app/components/globalDrawer/components.tsx

```diff
@@ -28,35 +36,193 @@ interface DrawerPanelProps {
+    if (savedWidth !== undefined) {
+      const minWidthPercent = 30;
+      const maxWidthPercent = 90;
+      return Math.min(Math.max(savedWidth, minWidthPercent), maxWidthPercent);
+    }
+
+    if (drawerWidth) {
+      // If width is already in percentage, parse and clamp it
+      if (drawerWidth.endsWith('%')) {
+        const parsedPercent = parseFloat(drawerWidth);
+        const minWidthPercent = 30;
+        const maxWidthPercent = 90;
+        return Math.min(Math.max(parsedPercent, minWidthPercent), maxWidthPercent);
+      }
+      // If width is in pixels, convert to percentage and clamp
+
+      const viewportWidth = typeof window === 'undefined' ? 1000 : window.innerWidth;
+      const parsedPixels = parseFloat(drawerWidth);
+      const percentValue = (parsedPixels / viewportWidth) * 100;
+      const minWidthPercent = 30;
+      const maxWidthPercent = 90;
+      return Math.min(Math.max(percentValue, minWidthPercent), maxWidthPercent);
+    }
+
+    return 50; // Default to 50%
+  };
+
+  // Store persisted width in localStorage, but don't use it for rendering state
+  const [persistedWidthPercent, setPersistedWidthPercent] =
+    useSyncedLocalStorageState<number>(
+      drawerKey ? getDrawerWidthKey(drawerKey) : 'drawer-width:default',
+      (value?: unknown) => {
+        const savedWidth = typeof value === 'number' ? value : undefined;
+        return calculateInitialWidth(savedWidth);
+      }
+    );
+
+  useLayoutEffect(() => {
+    if (panelRef.current) {
+      panelRef.current.style.width = `${persistedWidthPercent}%`;
+    }
+  }, [persistedWidthPercent, drawerKey, drawerWidth]);
+
+  const handleResizeStart = useCallback(
+    (e: React.MouseEvent) => {
+      e.preventDefault();
+
+      const handle = resizeHandleRef.current;
+      const panel = panelRef.current;
+      if (!handle || !panel) {
+        return;
+      }
+
+      // Mark as resizing
+      isResizingRef.current = true;
+      handle.setAttribute('data-resizing', 'true');
+      panel.classList.add('resizing');
+      initialMousePositionRef.current = e.clientX;
+
+      const viewportWidth = window.innerWidth;
+      const minWidthPercent = 30;
+      const maxWidthPercent = 85;
+
+      const handleMouseMove = (moveEvent: MouseEvent) => {
+        moveEvent.preventDefault();
+
+        if (rafIdRef.current !== null) {
+          window.cancelAnimationFrame(rafIdRef.current);
+        }
+
+        rafIdRef.current = window.requestAnimationFrame(() => {
+          if (!panel || !handle || initialMousePositionRef.current === null) {
+            return;
+          }
+
+          const newWidthPercent =
+            ((viewportWidth - moveEvent.clientX) / viewportWidth) * 100;
+          const clampedWidthPercent = Math.min(
+            Math.max(newWidthPercent, minWidthPercent),
+            maxWidthPercent
+          );
+
+          panel.style.width = `${clampedWidthPercent}%`;
```

Comment:
Another way to do this that you might consider is by using css variables with `setProperty`

https://github.com/getsentry/sentry/blob/89606904d5e4f8b6272a0badbc4cd0568632d7b5/static/app/components/checkInTimeline/timelineCursor.tsx#L54-L96

you could set the width as a property and read it in the CSS. You could then do the clamping in the CSS actually. it might simplify this code a fair but. It does make the 'at min' and 'at max' more difficult, but I would argue that's mostly not important